experimental/dds/tree/src/test/EditLog.tests.ts

```diff
@@ -244,4 +244,108 @@ describe('EditLog', () => {
 
 		expect(log0.equals(differentLog)).to.be.false;
 	});
+
+	it('can configure a maximum size and evict edits once it grows larger than that size', () => {
+		const editLogSize = 10;
+		const log = new EditLog(undefined, undefined, undefined, editLogSize, editLogSize * 2);
+		const ids: EditId[] = [];
+
+		let editsEvicted = 0;
+
+		log.registerEditEvictionHandler((editsToEvict) => {
+			editsEvicted += editsToEvict;
+		});
+
+		for (let i = 0; i < editLogSize; i++) {
+			const edit = newEdit([]);
+			log.addSequencedEdit(edit, {
+				sequenceNumber: i,
+				referenceSequenceNumber: i - 1,
+			});
+			ids.push(edit.id);
+			expect(log.getIndexOfId(edit.id)).equals(i);
+		}
+		expect(log.length)
+			.equals(log.numberOfSequencedEdits)
+			.and.equals(editLogSize, 'Only sequenced edits should be present.');
+
+		const newIds: EditId[] = [];
+		for (let i = 0; i < editLogSize; i++) {
+			const edit = newEdit([]);
+			const sequenceNumber = editLogSize + 1;
+			log.addSequencedEdit(edit, {
+				sequenceNumber,
+				referenceSequenceNumber: sequenceNumber - 1,
+				minimumSequenceNumber: sequenceNumber - 1,
+			});
+			newIds.push(edit.id);
+			expect(log.getIndexOfId(edit.id)).equals(editLogSize + i);
+		}
+
+		expect(log.editIds).to.deep.equal(newIds, 'Edit IDs should have been evicted.');
+		expect(log.length).equals(log.numberOfSequencedEdits).and.equals(editLogSize, 'Edits should have been evicted');
+		expect(editsEvicted).to.equal(editLogSize);
+	});
+
+	describe('does not evict edits in the collaboration window', () => {
+		[0, 2, 8, 23, 50, 68, 255].forEach((startSequenceNumber) => {
+			[1, 7, 10, 13, 52].forEach((editLogSize) => {
+				[2, 15, 21, Math.floor(editLogSize * 1.5), editLogSize * 2].forEach((collaborationWindowSize) => {
+					it(`when accepting edits starting from sequence number ${startSequenceNumber} and targeting an edit log size of ${editLogSize} and a collaboration window size of ${collaborationWindowSize}`, () => {
+						const log = new EditLog(undefined, undefined, undefined, editLogSize, editLogSize * 2);
+						const ids: EditId[] = [];
+
+						let editsEvicted = 0;
+
+						log.registerEditEvictionHandler((editsToEvict) => {
+							editsEvicted += editsToEvict;
+						});
+
+						// Add enough edits to hit the target size
+						for (let i = 0; i < editLogSize; i++) {
+							const edit = newEdit([]);
+							log.addSequencedEdit(edit, {
+								sequenceNumber: startSequenceNumber + i,
```

Comment:
you could simulate "sparse" sequence numbers by having this go up by more than 1 sometimes.

Reply:
I added a separate test for sparse sequence numbers cause it made calculating expected numbers in this test really difficult.